app/controllers/posts_controller.rb

```diff
@@ -237,6 +238,20 @@ def destroy_many
     render nothing: true
   end
 
+  def merge_posts
+    params.require(:post_ids)
+
+    posts = Post.where(id: post_ids_including_replies).order(:id)
+    raise Discourse::InvalidParameters.new(:post_ids) if posts.blank?
```

Comment:
Hmm we can't just check for a blank here. Probably do a `map(&:id)` to make sure the ids are equal the `post_ids`. Otherwise, there might be cases where one of the id is valid we will not know about it.